### PR TITLE
feature/34-add-api-test-utilities-and-request-specification

### PR DESCRIPTION
**Pull request by lukewalker85**
### Summary
Added reusable API test utilities for request, response specifications and auth tokens.

## Changes  
- Added `RequestSpecificationFactoryTest.java` to test that base uri, content type and accept type are set. Also tests uri cannot be null or empty.
- Added `AuthTokenManagerTest.java` to test that arguments cannot be null or empty. 
- Added `RequestSpecificationFactory.java` to build request spec.
- Added `ResponseSpecificationFactory.java` to build response spec.
- Added `AuthTokenManager.java` to retrieve auth token.
- Updated `ci.yml` to contain api auth credentials (Credentials added to GitHub secrets).
- Updated `config.properties` with comment about where api credentials are stored. 

### Testing 
- `mvn verify`: 24 unit tests (Surefire), 28 integration tests (Failsafe) — all green
-  7 new unit tests in `RequestSpecificationFactoryTest.java`

closes #34

## Release Notes

* **Tests**
  * Added API authentication token retrieval utilities with comprehensive validation tests.
  * Added request and response specification factories for enhanced API testing.

* **Chores**
  * Updated CI workflow with API authentication environment variables.
  * Extended configuration with API base URI settings.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,4 +24,6 @@ jobs:
           LOCKED_OUT_USER: ${{ secrets.LOCKED_OUT_USER }}
           PASSWORD: ${{ secrets.PASSWORD }}
           INVALID_PASSWORD: ${{ secrets.INVALID_PASSWORD }}
+          API_USERNAME: ${{ secrets.API_USERNAME }}
+          API_PASSWORD: ${{ secrets.API_USERNAME }}
```
[the remark this answers is not in the document]
Comment (lukewalker85) [reply .github/workflows/ci.yml]: Fixed, thanks

File: src/test/java/com/automation/api/AuthTokenManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  @Test
+  void shouldThrowIllegalArgumentExceptionIfUsernameIsEmpty() {
+    assertThatThrownBy(
+            () ->
+                AuthTokenManager.getAuthToken(
+                    "", "", RequestSpecificationFactory.buildRequestSpec("https://example.com")))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
```
[the remark this answers is not in the document]
Comment (lukewalker85) [reply src/test/java/com/automation/api/AuthTokenManagerTest.java:28]: Thanks but the additional checks feel redundant the check for if null or empty is in the same code block.

File: src/test/java/com/automation/api/ResponseSpecificationFactory.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  private ResponseSpecificationFactory() {
+    // Utility class - prevent instantiation
+  }
+
+  /** Builds a Response Specification that contains a response code and content type */
+  public static ResponseSpecification buildResponseSpec(int responseCode) {
+    LOG.info("Creating Response Specification, response code: {}", responseCode);
+    ResponseSpecification spec =
+        RestAssured.expect().statusCode(responseCode).contentType(ContentType.JSON).log().all();
```
[the remark this answers is not in the document]
Comment (lukewalker85) [reply src/test/java/com/automation/api/ResponseSpecificationFactory.java]: Removed
